tools/system_libs.py

```diff
@@ -524,15 +510,18 @@ def build_objects(self, build_dir):
           object_uuid += 1
           o = os.path.join(build_dir, f'{object_basename}__{object_uuid}.o')
         commands.append(cmd + [src, '-o', o])
+        objects.add(o)
       elif batch_inputs:
         # Use relative paths to reduce the length of the command line.
         # This allows to avoid switching to a response file as often.
         src = os.path.relpath(src, build_dir)
         src = utils.normalize_path(src)
         batches.setdefault(tuple(cmd), []).append(src)
+        # No -o in command, use original file name.
+        objects.add(os.path.join(build_dir, shared.unsuffixed_basename(src) + '.o'))
```

Comment:
Can we set `o = os.path.join(build_dir, shared.unsuffixed_basename(src) + '.o')` here and continue to use a single `object.add(o)` below?